adapter/internal/discovery/xds/server.go

```diff
@@ -466,6 +520,109 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, deployedEnvironments []*
 		orgIDOpenAPIRoutesMap[organizationID] = routesMap
 	}
 
+	apiName := mgwSwagger.GetTitle()
+	apiVersion := mgwSwagger.GetVersion()
+	apiRangeIdentifier := GenerateIdentifierForAPIWithoutVersion(vHost, apiName)
+
+	_, _, _, err = ValidateAndGetVersionComponents(apiVersion)
+
+	isIntelligentRoutingEnabled := conf.Adapter.IsIntelligentRoutingEnabled
+	if isIntelligentRoutingEnabled && err == nil {
+		// Check the major and minor version ranges of the current API
+		existingMajorRangeLatestVersion, isMajorRangeRegexAvailable :=
+			orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier][GetMajorVersionRange(apiVersion)]
+		existingMinorRangeLatestVersion, isMinorRangeRegexAvailable :=
+			orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier][GetMinorVersionRange(apiVersion)]
+
+		// Check whether the current API is the latest version in the major and minor version ranges
+		isLatestMajorVersion := !isMajorRangeRegexAvailable || existingMajorRangeLatestVersion <= apiVersion
```

Comment:
`existingMajorRangeLatestVersion <= apiVersion` Is this a string comparison?
I mean what if existingMajorRangeLatestVersion is 12 and apiVersion is 2, 12 <= 2

Reply:
Will fix by using integer components as suggested above

Reply:
Fixed with: https://github.com/wso2/product-microgateway/commit/10575c0498ec6037c245cc1473891adaad8ee27a